sys/net/application_layer/sock_dns/dns_cache.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2022 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup net_sock_dns
+ * @{
+ * @file
+ * @brief   DNS cache implementation
+ * @author  Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ * @}
+ */
+
+#include "checksum/fletcher32.h"
+#include "net/sock/dns.h"
+#include "dns_cache.h"
+#include "ztimer.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+static struct dns_cache_entry {
+    uint32_t hash;
+    uint32_t expires;
+    union {
+#if IS_ACTIVE(SOCK_HAS_IPV4)
+        ipv4_addr_t v4;
+#endif
+#if IS_ACTIVE(SOCK_HAS_IPV6)
+        ipv6_addr_t v6;
+#endif
+    } addr;
+    uint8_t len;
+} cache[CONFIG_DNS_CACHE_SIZE];
+
+static uint8_t _addr_len(int family)
+{
+    switch (family) {
+    case AF_INET:
+        return sizeof(ipv4_addr_t);
+    case AF_INET6:
+        return sizeof(ipv6_addr_t);
+    }
+    return 0;
+}
+
+int sock_dns_cache_query(const char *domain_name, void *addr_out, int family)
+{
+    uint32_t now = ztimer_now(ZTIMER_MSEC) / MS_PER_SEC;
```

Comment:
`RIOT/sys/net/application_layer/sock_dns/dns_cache.c:53:46: error: 'MS_PER_SEC' undeclared (first use in this function)`

Adding `#include "time_units.h"` fixes this.